apps/weather_status/lib/Controller/WeatherStatusController.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @copyright Copyright (c) 2020, Julien Veyssier
+ *
+ * @author Julien Veyssier <eneiluj@posteo.net>
+ *
+ * @license AGPL-3.0
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program. If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+namespace OCA\WeatherStatus\Controller;
+
+use OCA\WeatherStatus\Service\WeatherStatusService;
+use OCP\AppFramework\Http;
+use OCP\AppFramework\Http\DataResponse;
+use OCP\AppFramework\OCSController;
+use OCP\ILogger;
+use OCP\IRequest;
+
+class WeatherStatusController extends OCSController {
+
+	/** @var string */
+	private $userId;
+
+	/** @var ILogger */
+	private $logger;
+
+	/** @var WeatherStatusService */
+	private $service;
+
+	public function __construct(string $appName,
+								IRequest $request,
+								ILogger $logger,
+								WeatherStatusService $service,
+								string $userId) {
+		parent::__construct($appName, $request);
+		$this->userId = $userId;
+		$this->logger = $logger;
+		$this->service = $service;
+	}
+
+	/**
+	 * @NoAdminRequired
+	 *
+	 * @return DataResponse
+	 */
+	public function usePersonalAddress(): DataResponse {
+		return new DataResponse($this->service->usePersonalAddress());
+	}
+
+	/**
+	 * @NoAdminRequired
+	 *
+	 * @return DataResponse
+	 */
+	public function setMode($mode): DataResponse {
+		return new DataResponse($this->service->setMode($mode));
+	}
+
+	/**
+	 * @NoAdminRequired
+	 *
+	 * @return DataResponse
+	 */
+	public function setLocation($address, $lat, $lon): DataResponse {
```

Comment:
```suggestion
	public function setLocation(string $address, ?float $lat, ?float $lon): DataResponse {
```

also missing from the doc block